wls-ergebnismeldung-service/src/test/java/de/muenchen/oss/wahllokalsystem/ergebnismeldungservice/rest/begruendung/BegruendungControllerIntegrationTest.java

```diff
@@ -0,0 +1,168 @@
+
+    @Nested
+    class GetBegruendung {
+
+        @AfterEach
+        void setup() {
+            SecurityUtils.runWith(Authorities.REPOSITORY_DELETE_BEGRUENDUNG);
+            begruendungRepository.deleteAll();
+        }
+
+        @Test
+        @WithMockUser(authorities = { Authorities.SERVICE_GET_BEGRUENDUNG, Authorities.REPOSITORY_READ_BEGRUENDUNG })
+        void emptyResponse() throws Exception {
+            val request = get("/businessActions/begruendung/wahlbezirkID/21/LTW_BZW_A");
+
+            val response = api.perform(request).andExpect(status().isNoContent()).andReturn();
+
+            Assertions.assertThat(response.getResponse().getContentAsString()).isEmpty();
+        }
+
+        @Test
+        @WithMockUser(
+                authorities = { Authorities.SERVICE_GET_BEGRUENDUNG, Authorities.REPOSITORY_READ_BEGRUENDUNG,
+                        Authorities.REPOSITORY_WRITE_BEGRUENDUNG }
+        )
+        void dataFound() throws Exception {
+            val wahlbezirkID1 = "wahlbezirkID1";
+            val wahlbezirkID2 = "wahlbezirkID2";
+
+            val wahlID1 = "wahlID1";
+            val wahlID2 = "wahlID2";
+
+            val stapelart1 = Stapelart.LTW_BZW_A;
+            val stapelart2 = Stapelart.LTW_BZW_B;
+
+            val begruendung1 = new Begruendung();
+            begruendung1.setBezirkUndWahlIDStapelart(new BezirkUndWahlIDStapelart(wahlbezirkID1, wahlID1, stapelart1));
+            begruendung1.setGrund1("grund1");
+            begruendung1.setGrund2("grund2");
+            begruendung1.setUnstimmigkeiten(true);
+            begruendung1.setNachzaehlung(true);
+            begruendungRepository.save(begruendung1);
+
+            val begruendung2 = new Begruendung();
+            begruendung2.setBezirkUndWahlIDStapelart(new BezirkUndWahlIDStapelart(wahlbezirkID1, wahlID2, stapelart2));
+            begruendung2.setGrund1("grund1");
+            begruendungRepository.save(begruendung2);
+
+            val begruendung3 = new Begruendung();
+            begruendung3.setBezirkUndWahlIDStapelart(new BezirkUndWahlIDStapelart(wahlbezirkID2, wahlID1, stapelart2));
+            begruendung3.setGrund1("grund1");
+            begruendungRepository.save(begruendung3);
+
+            val expectedIDOfResponse = new BezirkUndWahlIDStapelart(wahlbezirkID1, wahlID1, stapelart1);
+            val expectedResponse = new BegruendungDTO(expectedIDOfResponse, "grund1", "grund2", true, true);
+
+            val request = get("/businessActions/begruendung/" + wahlbezirkID1 + "/" + wahlID1 + "/" + stapelart1);
+            val response = api.perform(request).andExpect(status().isOk()).andReturn();
+
+            val responseBody = objectMapper.readValue(response.getResponse().getContentAsString(), BegruendungDTO.class);
+            Assertions.assertThat(responseBody).usingRecursiveComparison().isEqualTo(expectedResponse);
+        }
+    }
+
+    @Nested
+    class AddBegruendung {
```

Comment:
```suggestion
    class PostBegruendung {
```
Klassenname sollte wie die Methode heißen